packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -463,9 +524,23 @@ class ContainerLayer extends Layer {
   /// their children using [addChildrenToScene], then reverse the aforementioned
   /// effects before returning from [addToScene].
   void addChildrenToScene(ui.SceneBuilder builder, [Offset childOffset = Offset.zero]) {
+    assert((){
+      if (childOffset != Offset.zero) {
+        debugPrint(
+          'WARNING: please use OffsetLayer to handle offset and do not '
+          'pass offsets down to children as it disables our retained '
+          'rendering optimization.'
+        );
```

Comment:
please either fail (throw a FlutterError; see https://master-docs-flutter-io.firebaseapp.com/flutter/foundation/FlutterError/message.html for details), or don't output anything. We don't do warnings, because people ignore them and you end up with lots of stuff in the logs making development harder.